src/pages/overviewCluster/overviewCluster.ts

```diff
@@ -0,0 +1,118 @@
+import { getBucketData } from "../../util/OverviewClusterUtils/ClusterOverviewBucketTab";
+import { getGeneraStorageDetails, getGeneralClusterDetails, getGeneralQuotaDetails, getGeneralRAMDetails } from "../../util/OverviewClusterUtils/ClusterOverviewGeneralTab";
+
+const fetchBucketNames = (bucketsSettings: BucketSettings[] | undefined, connection: IConnection): Array<Bucket> => {
+    let Buckets: Array<Bucket> = [];
+    if (bucketsSettings !== undefined) {
+        for (let bucketSettings of bucketsSettings) {
+            let bucketName: string = bucketSettings.name;
+            let bucket: Bucket | undefined = connection?.cluster?.bucket(bucketName);
+            if (bucket !== undefined) {
+                Buckets.push(bucket);
+            }
+        }
+    }
+    return Buckets;
+};
+
+export async function fetchClusterOverview(node: ClusterConnectionNode, context: vscode.ExtensionContext) {
+
+    const connection = Memory.state.get<IConnection>("activeConnection");
+
+    if (!connection) {
+        return;
+    }
+
+    const currentPanel = vscode.window.createWebviewPanel(
+        "clusterOverview",
+        "Cluster Overview",
+        vscode.ViewColumn.One,
+        {
+            enableScripts: true,
+            enableForms: true,
+        }
+    );
+
+    currentPanel.iconPath = vscode.Uri.file(
+        path.join(__filename, "..", "..", "images", "cb-logo-icon.svg")
+    );
+    currentPanel.webview.html = `Loading....`;
+    let ClusterOverviewObject: IClusterOverview = {
+        GeneralDetails: null,
+        Buckets: null,
+        Nodes: null,
+        Title: '',
+        BucketsHTML: [],
+        NodesHTML: []
+    };
+
+    try {
+        // Fetch server overview details
+        const restAPIObject = new CouchbaseRestAPI(connection);
+        const serverOverview: ServerOverview | undefined = await restAPIObject.getOverview();
+
+        // Fetch Buckets
+        let bucketsSettings = await connection?.cluster?.buckets().getAllBuckets();
+        let Buckets = fetchBucketNames(bucketsSettings, connection);
+
+        // General Overview
+        let generalClusterDetails = getGeneralClusterDetails(serverOverview);
+        let generalQuotaDetails = getGeneralQuotaDetails(serverOverview);
+        let generalRAMDetails = getGeneralRAMDetails(serverOverview);
+        let generalStorageDetails = getGeneraStorageDetails(serverOverview);
+
+        // Buckets Data
+        let bucketsHTML: IKeyValuePair[] = [];
+        for (let bucket of Buckets) {
+            const bucketOverview: BucketOverview | undefined = await restAPIObject.getBucketsOverview(bucket.name);
+            let bucketHTML = bucketOverview !== undefined ? getBucketData(bucketOverview) : '';
+            bucketsHTML.push({ key: bucket.name, value: bucketHTML });
+        }
+
+        // Nodes Data
+        const NodesHTML: IKeyValuePair[] = serverOverview !== undefined ? getNodeTabData(serverOverview) : [];
+
+        ClusterOverviewObject = {
```

Comment:
camelCase problem please fix
Keep it clusterOverviewObject
and fix whole object